tests/providers/teradata/hooks/test_teradata.py

```diff
@@ -291,3 +291,59 @@ def getvalue(self):
         self.cur.fetchall.return_value = {k: bindvar(v) for k, v in parameters.items()}
         result = self.test_db_hook.callproc("proc", True, parameters)
         assert result == parameters
+
+    def test_set_query_band(self):
+        query_band_text = "example_query_band_text"
+        _handle_user_query_band_text(query_band_text)
+        self.test_db_hook.set_query_band(query_band_text, self.conn)
+        self.conn.cursor.assert_called_once()
+
+    @mock.patch("teradatasql.connect")
+    def test_query_band_not_in_conn_config(self, mock_connect):
+        extravalues = {"query_band": "appname=airflow;org=test;"}
+        self.connection.extra = json.dumps(extravalues)
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        args, kwargs = mock_connect.call_args
+        assert args == ()
+        assert kwargs["host"] == "host"
+        assert kwargs["database"] == "schema"
+        assert kwargs["dbs_port"] == "1025"
+        assert kwargs["user"] == "login"
+        assert kwargs["password"] == "password"
+        assert "query_band" not in kwargs
+
+
+def test_handle_user_query_band_text_invalid():
+    query_band_text = _handle_user_query_band_text("invalid_queryband")
+    assert query_band_text == "invalid_queryband;appname=airflow;org=teradata-internal-telem;"
+
+
+def test_handle_user_query_band_text_override_appname():
+    query_band_text = _handle_user_query_band_text("appname=test;")
+    assert query_band_text == "appname=test_airflow;org=teradata-internal-telem;"
+
+
+def test_handle_user_query_band_text_append_org():
+    query_band_text = _handle_user_query_band_text("appname=airflow;")
+    assert query_band_text == "appname=airflow;org=teradata-internal-telem;"
+
+
+def test_handle_user_query_band_text_user_org():
+    query_band_text = _handle_user_query_band_text("appname=airflow;org=test")
+    assert query_band_text == "appname=airflow;org=test"
+
+
+def test_handle_user_query_band_text_none():
+    query_band_text = _handle_user_query_band_text(None)
+    assert query_band_text == "appname=airflow;org=teradata-internal-telem;"
```

Comment:
I think these test cases fail now.
Because we have changed the order like this "org=teradata-internal-telem;appname=airflow;"